### DISCUSSION TITLE
Return git's file hash to retrieve full content

### DISCUSSION THREAD

**Discussion by tawfikb-pdoc** [zricethezav/gitleaks#589] [Q&A]
Hi !
In order to retrieve a file content (at a given commit), I thought about retrieving the file hash and call the `git cat-file -p $hash` command.
Do you think this would make sense, or is there a better way of doing this ?
This is different from the feature proposed here https://github.com/zricethezav/gitleaks/pull/555, since in my case I would need the "git hash" content, not the hash of just the file's content.
I already implemented this locally, so I'm ready to issue a PR if it makes sense.

**Comment by w0rmr1d3r** [comment]
Hello @tawfikb-pdoc !

Open that PR and let's take a look :)

Cheers!

**Comment by Maitham16** [comment]
Your Approach (git cat-file -p $hash)
Requires retrieving the file’s hash first using:
git ls-tree -r <commit> | grep "<filename>"

git show <commit>:<file>
- Directly retrieves the file content at a given commit.
- Simpler and avoids unnecessary intermediate steps.
If your implementation requires working directly with Git hashes for a specific reason (e.g., for another feature that requires object hashes), your approach could still be justified. However, if the goal is just to retrieve file content, git show is the more efficient and idiomatic solution.

If you proceed with a PR, consider whether working with the Git object hash provides additional benefits beyond what git show already offers. If not, using git show might be the preferred approach.

**Reply by ghost** [reply]
It works
